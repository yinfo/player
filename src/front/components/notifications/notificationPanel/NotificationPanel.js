// @flow

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

type Props = {
  title?: string,
  children: any,
};

const NotificationBody = styled.div`
  height: 370px;
  overflow-y: scroll;
`;

const NotificationPanel = ({ title = 'Notifications', children }: Props) => (
  <section className="panel">
    <header className="panel-heading">{title}</header>
    <NotificationBody className="panel-body" id="noti-box">
      {children}
    </NotificationBody>
  </section>
);

NotificationPanel.displayName = 'NotificationPanel';

NotificationPanel.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node.isRequired,
};

export default NotificationPanel;
