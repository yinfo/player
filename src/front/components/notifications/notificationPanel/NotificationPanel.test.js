// @flow

import React from 'react';
import { shallow } from 'enzyme';
import NotificationPanel from './NotificationPanel';

describe('NotificationPanel component', () => {
  const props = {
    title: 'notification panel title',
  };

  it('renders as expected', () => {
    const component = shallow(
      <NotificationPanel {...props}>
        <p>a NotificationPanel</p>
      </NotificationPanel>,
    );
    expect(component).toMatchSnapshot();
  });
});
