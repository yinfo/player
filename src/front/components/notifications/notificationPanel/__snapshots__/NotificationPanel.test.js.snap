// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationPanel component renders as expected 1`] = `ShallowWrapper {}`;
